### DISCUSSION TITLE
gitlab Exploit failed: ArgumentError wrong number of arguments (given 3, expected 1..2)

### DISCUSSION THREAD

**Discussion by myusrn** [rapid7/metasploit-framework#15455] [Q&A] [answered]
msfconsole | msf6 > version -> 6.0.53-dev or 6.0.54-dev 

msf6 > search gitlab | use 1 / exploit/multi/http/gitlab_file_read_rce 

msf6 > set rhosts <remote gitlab host ip> + set lhost <localhost ip> + set username <gitlab non-root user> + set password <gitlab non-root user password> 

msf6 > run -> [-] Exploit failed: ArgumentError wrong number of arguments (given 3, expected 1..2)

this result doesn't arise on system that had metasploit-framework version 6.0.45-dev and it does arise when its been update to current kali-linux 6.0.53-dev and 6.0.54-dev package version.

**Comment by myusrn** [comment, marked as answer]
See https://github.com/rapid7/metasploit-framework/issues/15456 which is proposing that this is a bug so here it would be interesting to discuss if anyone has found an interim workaround.

**Reply by bcoles** [reply]
This was resolved in https://github.com/rapid7/metasploit-framework/pull/15500.
